pkg/doc/signature/ed25519signature2018/suite.go

```diff
@@ -81,18 +93,28 @@ func (s *SignatureSuite) GetDigest(doc []byte) []byte {
 }
 
 // Verify will verify a signature.
-func (s *SignatureSuite) Verify(pubKey *verifier.PublicKey, doc, signature []byte) error {
-	// ed25519 panics if key size is wrong
-	if l := len(pubKey.Value); l != ed25519.PublicKeySize {
-		return errors.New("ed25519: bad public key length")
+func (s *SignatureSuite) Verify(pubKeyValue interface{}, doc, signature []byte) error {
+	switch pubKey := pubKeyValue.(type) {
+	case *verifier.PublicKey:
+		// ed25519 panics if key size is wrong
+		if l := len(pubKey.Value); l != ed25519.PublicKeySize {
+			return errors.New("ed25519: bad public key length")
+		}
+
+		verified := ed25519.Verify(pubKey.Value, doc, signature)
+		if !verified {
+			return errors.New("signature doesn't match")
+		}
+
+		return nil
+
```

Comment:
I assume `case *keyset.Handle:` is coming up in another PR?

Reply:
I am basing on a more generic Crypto API interface which takes `kh interface{}`. In this way, not only **tinkrypto** is supported but any other crypto implementation as well. That's my idea.

Reply:
I'm ok with generic SignatureSuite API.

Similarly to KMS and Crypto APIs, they offer generic functions. The default implementation is using Tink (`*keyset.Handle`) as argument types.

same here, there should be a case for Tink's *keyset.Handle to call aries-framwork-go's default Crypto.Verify() instead of directly calling de25519 with []byte key

Reply:
It is what is happening in a `default` of the `switch`: we use Crypto instance passed as a functional argument to verify a signature. I think there is no need to introduce a direct dependency on `tinkcrypto` here.

Reply:
OK, we agreed with @Baha that can use `localkms.PubKeyBytesToHandle()` to convert public key bytes to key handle and use it with crypto. This will require some time to change, I will push the solution later on today.

Reply:
I think we should also check a type of the crypto impl. It it's a `tinkcrypto`, which can convert public key bytes to `*keyset.Handle` and call crypto.Verify(). In contrast, if other crypto impl is used then it could not accept `*keyset.Handle`, in this case, we will get into trouble. What do you think? @Baha-sk  @troyronda

Reply:
this is why `case *keyset.Handle:` will manage Tink wrapped keys, other implementations can support `case verify.PublicKey`

The API will be flexible with these two case conditions.